ostree-fdo-container.sh

```diff
@@ -0,0 +1,881 @@
+    if [[ $RESULTS == 1 ]]; then
+        greenprint "💚 Success"
+    else
+        greenprint "❌ Failed"
+        clean_up
+        exit 1
+    fi
+}
+
+###########################################################
+##
+## Prepare edge prod and stage repo
+##
+###########################################################
+# Have a clean prod repo
+greenprint "🔧 Prepare edge prod repo"
+sudo rm -rf "$PROD_REPO"
+sudo mkdir -p "$PROD_REPO"
+sudo ostree --repo="$PROD_REPO" init --mode=archive
+sudo ostree --repo="$PROD_REPO" remote add --no-gpg-verify edge-stage "$STAGE_REPO_URL"
+
+# Prepare stage repo network
+greenprint "🔧 Prepare stage repo network"
+sudo podman network inspect edge >/dev/null 2>&1 || sudo podman network create --driver=bridge --subnet=192.168.200.0/24 --gateway=192.168.200.254 edge
+
+# Clear container running env
+greenprint "🧹 Clearing container running env"
+# Remove any status containers if exist
+sudo podman ps -a -q --format "{{.ID}}" | sudo xargs --no-run-if-empty podman rm -f
+# Remove all images
+sudo podman rmi -f -a
+
+##########################################################
+##
+## Prepare FDO AIO container
+##
+##########################################################
+greenprint "🔧 Preparing fdo AIO server"
+sudo mkdir aio
+sudo podman run -v "$PWD"/aio/:/aio:z \
+  "quay.io/fido-fdo/aio:nightly" \
+  aio --directory aio generate-configs-and-keys --contact-hostname "$FDO_SERVER_ADDRESS"
+
+sudo /usr/local/bin/yq -iy '.service_info.diskencryption_clevis |= [{disk_label: "/dev/vda4", reencrypt: true, binding: {pin: "tpm2", config: "{}"}}]' aio/configs/serviceinfo_api_server.yml
+# FDO user does not have password, use ssh key and no sudo password instead
+sudo /usr/local/bin/yq -iy '.service_info.initial_user |= {username: "fdouser", sshkeys: ["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCzxo5dEcS+LDK/OFAfHo6740EyoDM8aYaCkBala0FnWfMMTOq7PQe04ahB0eFLS3IlQtK5bpgzxBdFGVqF6uT5z4hhaPjQec0G3+BD5Pxo6V+SxShKZo+ZNGU3HVrF9p2V7QH0YFQj5B8F6AicA3fYh2BVUFECTPuMpy5A52ufWu0r4xOFmbU7SIhRQRAQz2u4yjXqBsrpYptAvyzzoN4gjUhNnwOHSPsvFpWoBFkWmqn0ytgHg3Vv9DlHW+45P02QH1UFedXR2MqLnwRI30qqtaOkVS+9rE/dhnR+XPpHHG+hv2TgMDAuQ3IK7Ab5m/yCbN73cxFifH4LST0vVG3Jx45xn+GTeHHhfkAfBSCtya6191jixbqyovpRunCBKexI5cfRPtWOitM3m7Mq26r7LpobMM+oOLUm4p0KKNIthWcmK9tYwXWSuGGfUQ+Y8gt7E0G06ZGbCPHOrxJ8lYQqXsif04piONPA/c9Hq43O99KPNGShONCS9oPFdOLRT3U= ostree-image-test"]}' aio/configs/serviceinfo_api_server.yml
+# No sudo password required by ansible
+tee /tmp/fdouser > /dev/null << EOF
+fdouser ALL=(ALL) NOPASSWD: ALL
+EOF
+sudo /usr/local/bin/yq -iy '.service_info.files |= [{path: "/etc/sudoers.d/fdouser", source_path: "/tmp/fdouser"}]' aio/configs/serviceinfo_api_server.yml
+
+greenprint "🔧 Prepare fdo AIO manufacturing key and cert"
+DIUN_PUB_KEY_ROOT_CERTS=$(sudo cat aio/keys/diun_cert.pem)
+DIUN_PUB_KEY_HASH=sha256:$(sudo openssl x509 -fingerprint -sha256 -noout -in aio/keys/diun_cert.pem | cut -d"=" -f2 | sed 's/://g')
+
+greenprint "🔧 Starting fdo AIO server"
```

Comment:
This test only covered aio container. What about the other container? The aio is not for product use, it's for test and demo usage. What's your thoughts?

Reply:
Yes, actually I was thinking to cover standalone fdo servers/container but not only aio server/container, I plan to:
1. For fdo server testing, in upstream we use fdo aio server, so in downstream I plan to replace fdo aio server with standalone fdo manufacturing server, rendzvous server, owner server and service info server. In this way, we can cover both aio server and standalone fdo servers.
2. For fdo container testing, I will not use aio container, but use standalone fdo server containers instead.
What's your thoughts?